crates/accelerate/src/split_2q_unitaries.rs

```diff
@@ -19,20 +21,77 @@ use qiskit_circuit::circuit_instruction::OperationFromPython;
 use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType, Wire};
 use qiskit_circuit::imports::UNITARY_GATE;
 use qiskit_circuit::operations::{Operation, Param};
+use qiskit_circuit::packed_instruction::PackedInstruction;
+use qiskit_circuit::Qubit;
 
 use crate::two_qubit_decompose::{Specialization, TwoQubitWeylDecomposition};
 
+/// Extracts the K1r and K1l gates from the decomposition
+/// and creates them as new 1-qubit unitary gates.
+fn create_k1_gates<'a>(
+    decomp: &'a TwoQubitWeylDecomposition,
+    py: Python<'a>,
+) -> PyResult<(Bound<'a, PyAny>, Bound<'a, PyAny>)> {
+    let k1r_arr = decomp.K1r(py);
+    let k1l_arr = decomp.K1l(py);
+    let kwargs = PyDict::new(py);
+    kwargs.set_item(intern!(py, "num_qubits"), 1)?;
+    let k1r_gate = UNITARY_GATE
+        .get_bound(py)
+        .call((k1r_arr, py.None(), false), Some(&kwargs))?;
+    let k1l_gate = UNITARY_GATE
+        .get_bound(py)
+        .call((k1l_arr, py.None(), false), Some(&kwargs))?;
+    Ok((k1r_gate, k1l_gate))
```

Comment:
This actually needs an update now since #13759 I'll push up a small commit to fix it, since I caused the conflict.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13531/commits/0f84fc46abd2092521789b61996b28174061c9a3